internal/store/build_reason.go

```diff
@@ -0,0 +1,24 @@
+package store
+
+type BuildReason int
+
+const BuildReasonNone = BuildReason(0)
+
+const (
+	BuildReasonFlagFiles BuildReason = 1 << iota
+	BuildReasonFlagConfig
+	BuildReasonFlagCrash
+	BuildReasonFlagInit
+)
+
+func (r BuildReason) With(flag BuildReason) BuildReason {
+	return r | flag
+}
+
+func (r BuildReason) Has(flag BuildReason) bool {
+	return r&flag == flag
```

Comment:
what... what is this syntax? o_0

Reply:
bitwise &
https://golang.org/ref/spec#Arithmetic_operators

it means "is this bit set"